### ISSUE TITLE
With getComponent how to pass props?

### ISSUE THREAD

**Issue by AlastairTaft** [rackt/react-router#2488]
I can't figure out how to pass props to the component. This is important as I don't want to fetch data in the `componentDidMount` method as it will then be invisible to search engines.

My code looks like this

```
const router = 
  <Route path="/" component={App}>
    <IndexRoute onEnter={redirectToLogin} component={LandingPage} />
    <Route path="panel" component={ ControlPanel }>
      ...
    </Route>
    <Route 
      path="/:handle" 
      onEnter={maybeRedirectToHome}
      getComponent={(location, callback)=> {
          getSiteHandleByName(location.pathname.slice(1))
          .then(function(handle){
            if (handle){
              callback(null, Portfolio)
            } else {
              callback(null, NotFound)
            }
          })
          .catch(callback)
      }}
      getChildRoutes={(location, callback)=> {
        callback(null, portfolioRoutes)
      }} 
    />
  </Route>
```

I'm trying to serve up a portfolio React App when the user visits a valid url like `mysite.com/thishandleisvalid` but I'll need to also fetch all the content for that app at the `getComponent` point and pass it in as a property. E.g. you might normally do something like this `<Portfolio contentItems={fetchedItems} />`.

Is doing this possible?

**Comment by taion** [comment]
Thanks for your question!

We want to make sure that the GitHub issue tracker remains the best place to track bug reports and feature requests that affect the development of React Router.

Questions like yours deserve a purpose-built Q&A forum. Would you like to post this question to Stack Overflow with the tag #react-router? https://stackoverflow.com/questions/ask?tags=react-router.

We also have an active and helpful React Router community on Reactiflux, which is a great place to get fast help with React Router and with the rest of the React ecosystem. You can join at https://discord.gg/0ZcbPKXt5bYaNQ46.

**Comment by AlastairTaft** [comment]
Thanks, I've posted [a question](https://stackoverflow.com/questions/33571734/with-getcomponent-how-to-pass-props) on stackoverflow.

**Comment by AlastairTaft** [comment]
I've highlighted a use case that `react-router` doesn't support, see [the question](https://stackoverflow.com/questions/33571734/with-getcomponent-how-to-pass-props/33578098#33578098) for more info. Might be a useful feature to be able to customise component props? 

Could have a way of changing the nextState object to add values to props. As a work around I've hijacked the params property. But this makes my components tightly coupled to the router, which isn't great.
